transport/owr_transport_agent.c

```diff
@@ -310,6 +311,25 @@ static gboolean bus_call(GstBus *bus, GstMessage *msg, gpointer user_data)
         g_warn_if_fail(change_status != GST_STATE_CHANGE_FAILURE);
         break;
 
+    case GST_MESSAGE_QOS:
+    {
+        /* vtdec emits QoS whenever it cannot decode a frame. We translate this
+         * to a force key unit event as it normally means the frame data was broken. */
+        GstElement *decoder = GST_IS_ELEMENT(msg) ? GST_ELEMENT(GST_MESSAGE_SRC(msg)) : NULL;
+        if (decoder) {
+            GstElementFactory *factory = gst_element_get_factory(decoder);
+            if (g_str_has_prefix(gst_plugin_feature_get_name(factory), "vtdec")) {
+                guint64 processed = 0, dropped = 0;
+
+                gst_message_parse_qos_stats(msg, NULL, &processed, &dropped);
+                GST_WARNING("Video frame dropped - processed/dropped: %" G_GUINT64_FORMAT "/%" G_GUINT64_FORMAT, processed, dropped);
+
+                gst_element_send_event(decoder, gst_video_event_new_upstream_force_key_unit(GST_CLOCK_TIME_NONE, FALSE, 0));
```

Comment:
This doesn't seem like a good idea to me. vtdec might also drop frames if it was too slow with decoding for example... and requesting a keyframe then is not really going to help and might make everything worse :)

We currently don't have a notification when an element failed to decode something, only inside the element this can currently be detected and handled.